src/app/credExplorer/pagerankTable/aggregation.test.js

```diff
@@ -1,7 +1,422 @@
+    });
+    it("summarizes the group size appropriately", () => {
+      const {
+        nodeTypesArray,
+        edgeTypesArray,
+        scoredConnectionsArray,
+      } = example();
+      const aggregations = aggregateByConnectionType(
+        scoredConnectionsArray,
+        nodeTypesArray,
+        edgeTypesArray
+      );
+      for (const aggregation of aggregations) {
+        let empiricalSize = 0;
+        for (const nodeAggregation of aggregation.nodeAggregations) {
+          empiricalSize += nodeAggregation.summary.size;
+        }
+        expect(aggregation.summary.size).toEqual(empiricalSize);
+      }
+    });
+    it("summarizes the group score appropriately", () => {
+      const {
+        nodeTypesArray,
+        edgeTypesArray,
+        scoredConnectionsArray,
+      } = example();
+      const aggregations = aggregateByConnectionType(
+        scoredConnectionsArray,
+        nodeTypesArray,
+        edgeTypesArray
+      );
+      for (const aggregation of aggregations) {
+        let empiricalScore = 0;
+        for (const nodeAggregation of aggregation.nodeAggregations) {
+          empiricalScore += nodeAggregation.summary.score;
+        }
+        expect(aggregation.summary.score).toEqual(empiricalScore);
+      }
+    });
+    it("does not create empty groups", () => {
+      const {
+        nodeTypesArray,
+        edgeTypesArray,
+        scoredConnectionsArray,
+      } = example();
+      const aggregations = aggregateByConnectionType(
+        scoredConnectionsArray,
+        nodeTypesArray,
+        edgeTypesArray
+      );
+      for (const aggregation of aggregations) {
+        expect(aggregation.summary.size).not.toEqual(0);
+      }
+    });
+    it("handles the case where there are no connections", () => {
+      const {nodeTypesArray, edgeTypesArray} = example();
+      const aggregations = aggregateByConnectionType(
+        [],
+        nodeTypesArray,
+        edgeTypesArray
+      );
+      expect(aggregations).toHaveLength(0);
+    });
+    it("errors if any connection has no matching type", () => {
+      const {scoredConnectionsArray, nodeTypesArray} = example();
+      const shouldFail = () =>
+        aggregateByConnectionType(scoredConnectionsArray, nodeTypesArray, []);
+      expect(shouldFail).toThrowError("No matching EdgeType");
+    });
+    it("sorts the aggregations by total score", () => {
+      const {
+        nodeTypesArray,
+        edgeTypesArray,
+        scoredConnectionsArray,
+      } = example();
+      const aggregations = aggregateByConnectionType(
+        scoredConnectionsArray,
+        nodeTypesArray,
+        edgeTypesArray
+      );
+      let lastSeenScore = Infinity;
+      for (const aggregation of aggregations) {
+        const score = aggregation.summary.score;
+        expect(lastSeenScore >= score).toBe(true);
+        lastSeenScore = score;
+      }
+    });
+    it("within each aggregation, sub-aggregations are sorted by score", () => {
```

Comment:
Just to check: this is done by `aggregateByNodeType` and doubly tested,
right? (No objection.)

Reply:
Yep